js/preload/preloadMain.js

```diff
@@ -12,6 +12,7 @@
 // https://github.com/electron/electron/issues/2984
 //
 const { ipcRenderer, remote } = require('electron');
+const _ = require('underscore');
```

Comment:
i really want to avoid bringing in underscore dependency?  just for findindex.  we are using very modern JS here and you can use Array.findIndex here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex?v=example